internal/config.go

```diff
@@ -45,10 +44,16 @@ func registerConfig(config *hazelcast.Config, confPath string) error {
 	if out, err = yaml.Marshal(config); err != nil {
 		return err
 	}
+
+	filePath, _ := filepath.Split(confPath)
+	if err = os.MkdirAll(filePath, os.ModePerm); err != nil {
+		return fmt.Errorf("cannot create parent directories for configuration file: %w", err)
+	}
+
 	if err = ioutil.WriteFile(confPath, out, 0666); err != nil {
```

Comment:
This is not your code, but can you change `0666` to `0600`? So only the user has r/w access to it?